control-plane/api/v1alpha1/peeringacceptor_types.go

```diff
@@ -0,0 +1,82 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+func init() {
+	SchemeBuilder.Register(&PeeringAcceptor{}, &PeeringAcceptorList{})
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// PeeringAcceptor is the Schema for the peeringacceptors API.
+type PeeringAcceptor struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   PeeringAcceptorSpec   `json:"spec,omitempty"`
+	Status PeeringAcceptorStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// PeeringAcceptorList contains a list of PeeringAcceptor.
+type PeeringAcceptorList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []PeeringAcceptor `json:"items"`
+}
+
+// PeeringAcceptorSpec defines the desired state of PeeringAcceptor.
+type PeeringAcceptorSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Peer describes the information needed to create a peering.
+	Peer *Peer `json:"peer"`
+}
+
+type Peer struct {
+	Secret *Secret `json:"secret,omitempty"`
+	//Generator *Generator `json:"generator,omitempty"`
+	//Requester *Requester `json:"requester,omitempty"`
+}
+type Generator struct {
+	Secret *Secret `json:"secret,omitempty"`
+}
+type Requester struct {
+	Secret *Secret `json:"secret,omitempty"`
+}
+type Secret struct {
+	Name    string `json:"name,omitempty"`
+	Key     string `json:"key,omitempty"`
+	Backend string `json:"backend,omitempty"`
+}
+
+// PeeringAcceptorStatus defines the observed state of PeeringAcceptor.
+type PeeringAcceptorStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// LastReconcileTime is the last time the resource was reconciled.
+	// +optional
+	LastReconcileTime *metav1.Time  `json:"lastReconcileTime,omitempty" description:"last time the resource was reconciled"`
+	Secret            *SecretStatus `json:"secret,omitempty"`
+}
+type SecretStatus struct {
+	// TODO(peering): add additional status fields
+	Name       string                  `json:"name,omitempty"`
+	Key        string                  `json:"key,omitempty"`
+	Backend    string                  `json:"backend,omitempty"`
+	LatestHash string                  `json:"latestHash,omitempty"`
+	Kubernetes *KubernetesSecretStatus `json:"kubernetes,omitempty"`
```

Comment:
hey sorry wasn't sure what you meant here

Reply:
Ahh got it! Also so far I'm not using this field, since the Name Key and Backend are all I need to get the secret and compare it to the spec, do you think its better practice to use the object reference instead of the secret.name field?

Reply:
sorry i didnt respond to this. i think using the object reference makes a lot of sense. it is a lot more detailed and specific about the secret that is being generated IMO.

Reply:
ahh, yeah, I should update these fields then. Not yet done, still working on it.

Reply:
Hey @thisisnotashwin so when I was looking at the object reference type, it says to prefer to not use it and use local types instead: https://github.com/kubernetes/api/blob/be84346886a48ff00fc676f9cb37f10e79dfd0da/core/v1/types.go#L5624

Reply:
I renamed the fields in the local type like the example in the godoc above, wdyt?

Reply:
aha! i think that makes a lot of sense. thanks for doing this research Nitya!! this is excellent!